osu.Game/Beatmaps/Drawables/DifficultyIcon.cs

```diff
@@ -1,36 +1,127 @@
-// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
 using System;
 using osu.Framework.Allocation;
 using osu.Framework.Extensions.Color4Extensions;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Cursor;
 using osu.Framework.Graphics.Effects;
 using osu.Framework.Graphics.Shapes;
 using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
 using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
 using osu.Game.Rulesets;
 using osuTK;
 using osuTK.Graphics;
 
 namespace osu.Game.Beatmaps.Drawables
 {
-    public class DifficultyIcon : DifficultyColouredContainer
+    public class DifficultyIcon : DifficultyColouredContainer, IHasCustomTooltip
     {
         private readonly RulesetInfo ruleset;
 
-        public DifficultyIcon(BeatmapInfo beatmap, RulesetInfo ruleset = null)
+        public DifficultyIcon(BeatmapInfo beatmap, RulesetInfo ruleset = null, Boolean shouldShowTooltip = false)
             : base(beatmap)
         {
             if (beatmap == null)
                 throw new ArgumentNullException(nameof(beatmap));
 
             this.ruleset = ruleset ?? beatmap.Ruleset;
+            TooltipText = shouldShowTooltip ? $"{beatmap.Version}${beatmap.StarDifficulty:0.##}" : String.Empty;
 
             Size = new Vector2(20);
         }
 
+        public string TooltipText { get; set; }
+
+        public ITooltip GetCustomTooltip() => new DifficultyIconTooltip(AccentColour);
+
+        public class DifficultyIconTooltip : VisibilityContainer, ITooltip
```

Comment:
```suggestion
        private class DifficultyIconTooltip : VisibilityContainer, ITooltip
```
Should be private since the only usage for it is inside DifficultyIcon